File: extension/deployment/src/main/java/com/example/ExtensionProcessor.java
package com.example;

import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
import io.quarkus.deployment.annotations.BuildStep;
import io.quarkus.deployment.builditem.FeatureBuildItem;

public class ExtensionProcessor {

    private static final String FEATURE = "example-extension";

    @BuildStep
    FeatureBuildItem feature() {
        return new FeatureBuildItem(FEATURE);
    }

    @BuildStep
    AdditionalBeanBuildItem registerBeans() {
        return AdditionalBeanBuildItem.builder()
                .addBeanClass(ExampleService.class)
                .build();
    }

}

File: extension/deployment/src/main/kotlin/com/example/ExtensionProcessor.kt
package com.example

import io.quarkus.arc.deployment.AdditionalBeanBuildItem
import io.quarkus.deployment.annotations.BuildStep
import io.quarkus.deployment.builditem.FeatureBuildItem

class ExtensionProcessor {
  @BuildStep
  fun feature(): FeatureBuildItem {
    return FeatureBuildItem(FEATURE)
  }

  @BuildStep
  fun registerBeans(): AdditionalBeanBuildItem {
    return AdditionalBeanBuildItem.builder()
      .addBeanClass(ExampleService::class.java)
      .build()
  }

  companion object {
    private const val FEATURE = "example-extension"
  }
}
